url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/ssamot/mcts/ucb/optimisation/LockingContinuousProblem.java

```diff
@@ -0,0 +1,54 @@
+package ssamot.mcts.ucb.optimisation;
+
+import java.util.concurrent.SynchronousQueue;
+
+
+public class LockingContinuousProblem extends ContinuousProblem {
+	//private final ReentrantLock lock = new ReentrantLock();
+	
+	private SynchronousQueue<Object> lock = new SynchronousQueue<Object>(true);
+
+	private double[] x;
+
+	private double score;
+	
+	private int N; 
+	
+	@Override
+	public double getFtarget() {
+		// TODO Auto-generated method stub
+		return 0;
+	}
+
+	public double[] getX() {
+		return x;
+	}
+	
+	public void setScore(double score) {
+		this.score = score; 
+		this.N++;
+	}
+	
+	public int getN() {
+		return N;
+	}
+
+	public void unlock() {
+		try {
+			lock.take();
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)